### PR TITLE
Add Progress Indicators

### PR DESCRIPTION
**Pull request by rahulon12**
- Add Progress Indicators to views
- Also iterate over course pages

### CODE REVIEW

**Comment by azooz2003-bit** [comment]
The UI is lovely, but can we fix how the loader comes up via an animation or making sure it appears instantly. You can see a slightly ugly delay

https://github.com/user-attachments/assets/ed9b843a-f048-4163-9eef-c043304ecba1

File: CanvasPlusPlayground/Features/People/PeopleCommonView.swift [hunk 1]
```diff
@@ -39,6 +33,12 @@ struct PeopleCommonView: View {
                     .font(.caption)
                 }
             }
+
+            Section {
+                ForEach(commonCourses, id: \.id) { course in
+                    Text(course.displayName)
+                }
+            }
```
[the remark this answers is not in the document]
Comment (rahulon12) [reply CanvasPlusPlayground/Features/People/PeopleCommonView.swift:41]: The git diff is very weird, but all I’ve done is moved the spinner from the footer of the bottom list to the footer of the top section

[the remark this answers is not in the document]
Comment (rahulon12) [reply CanvasPlusPlayground/Features/People/PeopleCommonView.swift:41]: I thought there was too much moving around in that version

**Comment by rahulon12** [comment]
> The UI is lovely, but can we fix how the loader comes up via an animation or making sure it appears instantly. You can see a slightly ugly delay
> 
>  CleanShot.2024-12-18.at.14.26.40.mp4

I've modified it such that the default initialized value is `true`, so it loads instantly.

**Pull request merged**